documentation/monitoring.md

```diff
@@ -0,0 +1,49 @@
+Monitoring
+------------
+
+New Relic is a SaaS tool for monitoring and alerting. Ocim supports using `New Relic Synthetics` and `New Relic Alerts` for monitoring the instances.
+
+On Alerts, policies can be created to control the alerting behaviour. Alerts has two main components:
+* **Alert Policies**: An alert policy is a group of one or more alert conditions. A policy has two settings that apply to all of its conditions: incident preference and notification channels. A policy should be created before adding conditions to it.
+* **Alert Conditions**: its a combination of monitored data source, such as a Synthetics Monitor, and thresholds that define the behavior that will be considered a violation. Synthetics only reports if the pings succeeded or not, so our threshold is to check if the ping failed.
+
+Each **Alert Policy** contains one or more **Alert Conditions** that can be configured to modify the alerting behaviour.
+
+For example, on Open edX instances, we create a alert policy with 1 alert per policy, and add an alert condition for each service (LMS, Studio, Preview).
+That way, if any service is down, the monitor will trigger an alert. But if there's more than one service failing pings, only a single alert will be fired, to prevent firing a massive amount of alert emails when an entire instance goes down or there are OVH/AWS network issues.
+
+The image below shows the concept behind Alerts.
+
+![New Relic Alerts](./images/newrelic-alerts.png)
+
+A more in deph explanation of how New Relic Alerts work can be found [here](https://docs.newrelic.com/docs/alerts/new-relic-alerts/getting-started/new-relic-alerts-concepts-workflow).
+
+## How it works on Ocim
+
+Ocim automatically sets up monitoring for each instance, and creates all the necessary NewRelic resources for alerting. It does that by running `.enable_monitoring()` on an instance during provisioning.
+
+Additionally, the monitoring set up methods can be called manually to disable monitoring or rebuild NewRelic assets changed due to changes in configuration.
+
+* `.enable_monitoring()`: Set ups resources on New Relic to enable monitoring, if the resources already exist, overwrite them with new settings.
```

Comment:
@giovannicimolin, it won't overwrite any resources. It uses the database to check if resources exist and does nothing if they do. Otherwise, it creates a fresh set of resources and doesn't check the New Relic API for any existing resources for the instance.

Reply:
@lgp171188 So it won't update existing resources if settings are changed?

Reply:
@giovannicimolin, the resources here correspond to a URL to be monitored and a list of addresses to be notified. As long as those do not change, the existing resources will be left unmodified. If the URL is changed, the existing monitors for the old URL are not automatically removed - this is important because we don't want to delete those automatically to avoid issues (cf. [this comment](https://github.com/open-craft/opencraft/commit/7fb071b4cf0e06d275094839f7c60a0d0f4f8d7c#diff-c6922995a704923a2f6da744ec24a91aR94))

Reply:
Ok, I added this to the documentation